src/copilot/app/components/indexing/implementations/haystack.py

```diff
@@ -0,0 +1,443 @@
+
+        Returns
+        -------
+        dict
+            A dictionary containing a message indicating the success of the indexing operation.
+        """
+        # Get HTML content from all sitemap URLs
+        content = await self.scraper.html_content_from_sitemap(sitemap_url)
+
+        # Convert HTML content to Document objects
+        documents = self.parser.convert_html_to_documents(content)
+
+        # Clean documents
+        documents = self.parser.clean_documents(documents["documents"])
+
+        # Split documents into chunks
+        chunks = self.parser.split_documents(documents["documents"])
+
+        # TO DO: refactor embedding logic to embed from documents (add from_documents method)
+        # Embed documents
+        str_chunks = [doc.content for doc in chunks["documents"]]
+        document_embeddings = self.embedding_client.embed_documents(str_chunks)
+        document_embeddings = [e.embedding for e in document_embeddings]
+        urls = [doc.meta["url"] for doc in chunks["documents"]]
+
+        # Upsert documents into VectorDB
+        for embedding, doc, url in zip(document_embeddings, str_chunks, urls):
+            date = datetime.now()
+            await queries.insert_rag(str(embedding), doc, url, date, date)
+
+        return {"content": f"{sitemap_url}: RAG data indexed successfully"}
+
+    async def index_pdfs_from_sitemap(self, sitemap_url: str, language: str = "de") -> dict:
+        """
+        Indexes PDF content from the given sitemap URL into the VectorDB.
+
+        Parameters
+        ----------
+        sitemap_url : str
+            The URL of the sitemap to index PDF content from.
+        language : str, optional
+            The language of the PDF content, by default "de".
+
+        Returns
+        -------
+        dict
+            A dictionary containing a message indicating the success of the indexing operation.
+        """
+        # Get PDF content from all sitemap memento links
+        content = await self.scraper.pdf_content_from_sitemap(sitemap_url)
+
+        # Convert PDF content to Document objects
+        documents = self.parser.convert_pdf_to_documents(content)
+
+        # Clean documents
+        documents = self.parser.clean_documents(documents["documents"])
+
+        # Split documents into chunks
+        chunks = self.parser.split_documents(documents["documents"])
+
+        # TO DO: refactor embedding logic to embed from documents (add from_documents method)
+        # Embed documents
+        str_chunks = [doc.content for doc in chunks["documents"]]
+        document_embeddings = self.embedding_client.embed_documents(str_chunks)
+        document_embeddings = [e.embedding for e in document_embeddings]
+        urls = [doc.meta["url"] for doc in chunks["documents"]]
+
+        # Upsert documents into VectorDB
+        for embedding, doc, url in zip(document_embeddings, str_chunks, urls):
+            date = datetime.now()
```

Comment:
datetime is done within the database already

Reply:
Removed date args for "created_at", "modified_at". Also changed the indexing/queries.py insert_rag() function to not intake these parameters:

```
async def insert_rag(embedding: str,
                     text: str,
                     url: str):
    conn = await get_db_connection()

    try:
        await conn.execute("""
            INSERT
            INTO embeddings (embedding, text, url)
            VALUES ($1, $2, $3)
        """, embedding, text, url)

    finally:
        await conn.close()
```